docs/Toy/toy/dialects/toy.py

```diff
@@ -369,6 +372,18 @@ def __init__(self, input: SSAValue | None = None):
         return super().__init__(operands=[input])
 
 
+class ReshapeOpHasCanonicalisationPatternsTrait(HasCanonicalisationPatternsTrait):
+    @classmethod
+    def get_canonicalization_patterns(cls) -> Iterator[RewritePattern]:
+        from ..rewrites.optimise_toy import (
+            FoldConstantReshapeOpPattern,
+            ReshapeReshapeOpPattern,
+        )
+
+        yield ReshapeReshapeOpPattern()
+        yield FoldConstantReshapeOpPattern()
```

Comment:
semi-nit: Do we want to write it like this, or with a list/tuple?
I feel that this kind of stuff is a bit weird control-flow wise (technically, you jump in the function, then jump out, then jump again).

Reply:
I don't feel super strongly about this, it's just a way to avoid creating a temporary tuple. If you think it would be significantly clearer to return a tuple instead I'm happy to change it.